developer/jsdoc-style-guide.md

```diff
@@ -0,0 +1,229 @@
+```js
+/**
+ * @file SortableTable is a React Component wrapper around {@link https://react-table.js.org} ReactTable. Anything functionality from ReactTable should be available in SortableTable out of the box, but may require styling. For more, see {@link https://react-table.js.org/#/story/readme ReactTable docs}
+ *
+ * @module SortableTable
+ * @extends Component
+ */
+```
+
+2. Document each method with `@property` and `@param` tags:
+
+```js
+/**
+  * renderComponent
+  * @method
+  * @summary React component for displaying a not-found view
+  * @param {Object} props - React PropTypes
+  * @property {String|Object} className - String className or `classnames` compatible object for the base wrapper
+  * @property {String|Object} contentClassName - String className or `classnames` compatible object for the content wrapper
+  * @property {String} i18nKeyMessage - i18n key for message
+  * @return {Node} React node containing not-found view
+  */
+ ```
+
+3. Document `propTypes` at the bottom, before the propTypes declaration using `@property` tags:
+```js
+/**
+  * @name SortableTable propTypes
+  * @type {propTypes}
+  * @param {Object} props - React PropTypes
+  * @property {Object} collection collection to get data from
+  * @property {Array} columnMetadata provides filtered columns with i18n headers
+  * @property {Array} data provides array of objects to be used in place of publication data
+  * @property {Number} defaultPageSize how many results per page
+  * @property {Boolean} filterType filter by table, column, or both
+  * @property {Array} filteredFields provides filtered columns, use columnMetadata instead
+  * @property {Boolean} isFilterable show / hide column filter
+  * @property {Boolean} isResizeable allow resizing of table columns
+  * @property {Boolean} isSortable allow column sorting
+  * @property {String} matchingResultsCount provides Count publication to get count from
+  * @property {Number} minRows minimum amount of rows to display in table
+  * @property {String} noDataMessage text to display when no data is available
+  * @property {Function} onRowClick provides function / action when clicking on row
+  * @property {String} publication provides publication to get Meteor data from
+  * @property {object} query provides query for publication filtering
+  * @property {Array} selectedRows provides selected rows in the table
+  * @property {Function} transform transform of collection for grid results
+  * @return {Array} React propTypes
+  */
+```
+![screen shot 2017-10-18 at 12 12 00 pm](https://user-images.githubusercontent.com/3673236/31737844-9aaf6fba-b3fd-11e7-9939-d148a4843e47.png)
+![screen shot 2017-10-18 at 12 12 09 pm](https://user-images.githubusercontent.com/3673236/31737845-9ac73c1c-b3fd-11e7-946b-ddd771e9e4e0.png)
+![screen shot 2017-10-18 at 12 12 12 pm](https://user-images.githubusercontent.com/3673236/31737881-bfc6e648-b3fd-11e7-9247-db56893e9273.png)
+
+## Document a one-file module
+
```

Comment:
@machikoyasuda let's add something here to specifically address when "inlining" a 3rd party package, e.g. code that was originally a Meteor package.